frontend/app/services/wsaddress-service.server.ts

```diff
@@ -0,0 +1,151 @@
+import { z } from 'zod';
+import moize from 'moize';
+import type { ParseWSAddressRequestDTOProps } from '~/dtos/parse-address-request-dto.server';
+import { ParseWSAddressRequestDTO } from '~/dtos/parse-address-request-dto.server';
+import { ParseWSAddressResponseDTO } from '~/dtos/parse-address-response-dto.server';
+import type { ValidateWSAddressRequestDTOProps } from '~/dtos/validate-address-request-dto.server';
+import { ValidateWSAddressRequestDTO } from '~/dtos/validate-address-request-dto.server';
+import { ValidateWSAddressResponseDTO } from '~/dtos/validate-address-response-dto.server';
+
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('wsaddress-service.server');
+
+const ParseWSAddressResponseSchema = z.object({
+  responseType: z.string().optional(),
+  addressLine: z.string().optional(),
+  city: z.string().optional(),
+  province: z.string().optional(),
+  postalCode: z.string().optional(),
+  streetNumberSuffix: z.string().optional(),
+  streetDirection: z.string().optional(),
+  unitType: z.string().optional(),
+  unitNumber: z.string().optional(),
+  serviceAreaName: z.string().optional(),
+  serviceAreaType: z.string().optional(),
+  serviceAreaQualifier: z.string().optional(),
+  cityLong: z.string().optional(),
+  cityShort: z.string().optional(),
+  deliveryInformation: z.string().optional(),
+  extraInformation: z.string().optional(),
+  statusCode: z.string().optional(),
+  canadaPostInformation: z.array(z.string()).optional(),
+  message: z.string().optional(),
+  addressType: z.string().optional(),
+  streetNumber: z.string().optional(),
+  streetName: z.string().optional(),
+  streetType: z.string().optional(),
+  serviceType: z.string().optional(),
+  serviceNumber: z.string().optional(),
+  country: z.string().optional(),
+  warnings: z.string().optional(),
+  functionalMessages: z.array(z.object({
+    action: z.string(),
+    message: z.string()
+    })
+  ).optional(),
+});
+
+const ValidateWSAddressResponseSchema = z.object({
+  responseType: z.string().optional(),
+  statusCode: z.string().optional(),
+  functionalMessages: z.array(z.object({
+      action: z.string(),
+      message: z.string()
+    })
+  ).optional(),
+  message: z.string().optional(),
+  warnings: z.string().optional(),
+});
```

Comment:
This doesn't appear to be doing any useful validation. Consider removing it or adding a comment explaining why it exists.

Reply:
Added a TODO to change the validation set once the wsaddress endpoint becomes available